src/utils/freeable.ts

```diff
@@ -0,0 +1,13 @@
+export interface Freeable {
+  free(): void;
+}
+
+export type FreeableBucket = Array<Freeable | undefined>;
```

Comment:
Make this so it's also nullable. I've seen cases on jpg where I had to add null also